ui/src/components/song/SongSearch.tsx

```diff
@@ -0,0 +1,199 @@
+import {
+  Container,
+  Typography,
+  Box,
+  Stack,
+  TextField,
+  Autocomplete,
+  Checkbox,
+  FormControl,
+  FormControlLabel,
+  FormGroup,
+  FormLabel,
+  Button,
+} from '@mui/material';
+import { SongSearchProps, SongCardProps } from '../../types/song';
+import { useState, useEffect, useCallback } from 'react';
+
+const SongSearch = (props: SongSearchProps) => {
+  const Songs: SongCardProps[] = props.songs;
+  const prevFilter = props.filterData;
+  const [tempoList, setTempoList] = useState<string[]>([]);
+  const [themesList, setThemesList] = useState<string[]>([]);
+  const [search, setSearch] = useState(prevFilter?.search ?? '');
+  const [tempo, setTempo] = useState<string[]>(prevFilter?.tempo ?? []);
+  const [themes, setThemes] = useState<string[]>(prevFilter?.themes ?? []);
+  const [displayResult, setDisplayResult] = useState(
+    prevFilter?.display ?? {
+      tempo: true,
+      themes: true,
+      lyricsPreview: true,
+      originalKey: true,
+      year: true,
+      code: true,
+    }
+  );
+
+  const handleTempoChange = (event: React.ChangeEvent<{}>, value: string[]) => {
+    setTempo(value);
+  };
+
+  const handleThemeChange = (event: React.ChangeEvent<{}>, value: string[]) => {
+    setThemes(value);
+  };
+
+  const handleDisplayChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setDisplayResult({
+      ...displayResult,
+      [event.target.name]: event.target.checked,
+    });
+  };
+
+  const handleSubmit = () => {
+    props.setFilterData({ search: search, tempo: tempo, themes: themes, display: displayResult });
+    props.onClose();
+  };
+
+  const getSelectOptions = useCallback(() => {
+    const tempTempo: string[] = [];
+    const tempThemes: string[] = [];
+    Songs &&
+      Songs.map((song) => {
+        tempTempo.push(...song.tempo);
+        tempThemes.push(...song.themes);
+        return 0;
+      });
+    setTempoList(Array.from(new Set(tempTempo)));
+    setThemesList(Array.from(new Set(tempThemes)));
+  }, [Songs]);
+
+  useEffect(() => {
+    getSelectOptions();
+  }, [getSelectOptions]);
+  return (
+    <>
+      <Container>
+        <Stack direction="row" display="flex" justifyContent="space-between">
+          <Typography
+            color="#4B50B4"
+            fontFamily="Roboto"
+            fontSize="20px"
+            fontStyle="normal"
+            fontWeight={700}
+            lineHeight="28px"
+          >
+            SEARCH SONGS
```

Comment:
<img width="270" alt="image" src="https://github.com/hmcc-global/psalted-2/assets/24607458/71757300-766a-4d3b-a4c9-f4ef57595089">

fontFamily="Roboto" doesnt seem to work